content/post/2019-04-29-cranlogs-2-1-1.Rmarkdown

```diff
@@ -0,0 +1,143 @@
+
+### `cranlogs`, how and why?
+
+### `cranlogs` data source and its quality
+
+[RStudio](http://www.rstudio.com) publishes the download logs from their
+CRAN package mirror daily at <http://cran-logs.rstudio.com>. The `cranlogs` package queries a [web API maintained by R-hub](https://github.com/r-hub/cranlogs.app#the-api-of-the-cran-downloads-database) that contains the daily download numbers for each package.
+
+The RStudio CRAN mirror is not the only CRAN mirror, but it’s a popular one: it’s the default choice for RStudio users. The actual number of downloads over all CRAN mirrors is unknown. Therefore, using numbers of downloads from that mirror is a _proxy_ for actual numbers of downloads, and when analysing it one has to assume the observed trends are representative of what happens via other mirrors.
+
+The data `cranlogs` queries has other limitations:
+
+* zero counts on days the RStudio server serving the log was down. 
+
+* count inflation due to CRAN-wide shared daily downloads
+
+* re-download/CRAN mirror spikes due to package updates
+
+The latter two downsides can be corrected by the use of the [`adjustedcranlogs` package ](https://github.com/tylermorganwall/adjustedcranlogs) by Tyler Morgan-Wall. Refer to that package documentation and code to find out more about it adjusts the download counts.
+
+### `cranlogs` use cases
+
+Now, _why_ would you even be interested in the download counts of packages and R? As a package maintainer, is it only egosurfing? 
+
+```{r, echo = FALSE}
+blogdown::shortcode("tweet", "1115959975359209472")
+```
+
+Probably not, such data can be useful:
+
+* As the developer, you can e.g. assess effect of promotion efforts, and maybe use proof of your package's popularity in career reports? (Note that it is perfectly fine to maintain a small niche package! `cranlogs` itself is no `ggplot2`. :wink:) 
+
+* As the potential user, you can use download counts to compare popularity of packages you're hesitating to use. Again, quality does not necessitate popularity but a higher usage can be a good sign, in particular a good sign of the probability to get help from other users.
+
+* You could also use download counts as a way to summarize the "most important" packages around a topic, like what Zev Ross did below with spatial packages (the code attached to the tweet, [available as a GitHub gist](https://gist.github.com/zross/b64657b05156d43cf81dbdd093df5c13), uses the list of packages of the [spatial CRAN task view](https://cran.r-project.org/web/views/Spatial.html), get their downloads with `cranlogs`, and sort them by popularity).
+
+```{r zevross, echo = FALSE}
+blogdown::shortcode("tweet", "1123204696141963264")
+```
+
```

Comment:
You can use it for ranking when searching for packages. E.g. pkgsearch reports it, from the service not using the R package:

```r
❯ x <- pkgsearch::ps("graphs")
❯ x
- "graphs" ------------------------------------ 337 packages in 0.009 seconds -
  #     package    version by                    @ title
  1 100 DiagrammeR 1.0.1   Richard Iannone      9d Graph/Network Visualization
  2  91 igraph     1.2.4.1 Gábor Csárdi         9d Network Analysis and Visu...
  3  62 huge       1.3.2   Haoming Jiang       23d High-Dimensional Undirect...
  4  62 ggraph     1.0.2   Thomas Lin Pedersen 10M An Implementation of Gram...
  5  59 networkD3  0.4     Christopher Gandrud  2y D3 JavaScript Network Gra...
  6  55 diagram    1.6.4   Karline Soetaert     2y Functions for Visualising...
  7  54 qgraph     1.6.1   Sacha Epskamp        3M Graph Plotting Methods, P...
  8  43 tidygraph  1.1.2   Thomas Lin Pedersen  2M A Tidy API for Graph Mani...
  9  34 plotly     4.9.0   Carson Sievert      21d Create Interactive Web Gr...
 10  32 showtext   0.6     Yixuan Qiu           4M Using Fonts More Easily i...

❯ tibble::as.tibble(x)$downloads_last_month
 [1]  23980 411677  15032  13917  14085   2685  10711   4282 134100   6810
```

Although it does not use it for ranking (yet), but it will. Anyway, you can mention this is you like.

Reply:
mentioned!